core/dbt/compilation.py

```diff
@@ -547,6 +547,8 @@ def compile_node(
         the node's raw_code into compiled_code, and then calls the
         recursive method to "prepend" the ctes.
         """
+        # REVIEW: UnitTestDefinition shouldn't be possible here because of the
+        # type of node, and it is likewise an invalid return type.
```

Comment:
I don't actually remember why this is here.  We went around a couple of times on different ways of handling unit tests, so my guess is that this might have come from an earlier iteration, before we switched to making a separate unit test manifest.  My recommendation is to remove it and see if any tests fail.